src/main/kotlin/it/pagopa/wallet/cdc/PaymentWalletsLogEventsStream.kt

```diff
@@ -44,22 +51,49 @@ class PaymentWalletsLogEventsStream(
                                 Aggregation.project(changeStreamOptionsConfig.project)
                             )
                         )
-                        .resumeAt(Instant.now())
+                        .resumeAt(redisResumePolicyService.getResumeTimestamp())
                         .build(),
                     BsonDocument::class.java
                 )
-                .flatMap {
-                    Mono.defer {
-                            walletPaymentCDCEventDispatcherService.dispatchEvent(
-                                it.raw?.fullDocument?.toBsonDocument()
-                            )
-                        }
-                        .onErrorResume {
-                            logger.error("Error during event handling : ", it)
-                            Mono.empty<BsonDocument>()
-                        }
-                }
+                // Process the elements of the Flux
+                .flatMap { processEvent(it) }
+                // Save resume token every n emitted elements
+                .index()
+                .flatMap { saveCdcResumeToken(it) }
```

Comment:
```suggestion
                .index { idx, document -> Pair(idx,document) }
                .flatMap {(idx,document)->  saveCdcResumeToken(idx,document) }
```
use parameter deconstructoring here in order to have explicit paramenters in the `saveCdcResumeToken` function

Reply:
Appliead in [6f724d6](https://github.com/pagopa/pagopa-payment-wallet-cdc-service/pull/11/commits/6f724d6a62d4c67307cc64fbf807e7eb10199470)